### PR TITLE
Fix Negative SKU Inventory Handling In Product Availability

### LINKED ISSUE #193: Negative inventory level of sku issue
Prestashop supports setting negative inventory level to sku. If one of the sku has negative inventory level, others have positive, and the summary is negative or zero, the availability should be instock, but nosto considers it out of stock.

### PR DESCRIPTION
**Pull request by supercid**
Closes #193

### CODE REVIEW

File: classes/models/NostoProduct.php [hunk 1]
```diff
@@ -255,7 +255,16 @@ protected static function checkAvailability(Product $product)
         if (!$product->active || $product->visibility === 'none') {
             return self::INVISIBLE;
         }
-        return ($product->checkQty(1)) ? self::IN_STOCK : self::OUT_OF_STOCK;
+        $combinations = $product->getAttributeCombinations();
+        if (empty($combinations)) {
```
Comment (chenliangde) [inline classes/models/NostoProduct.php]: how about "if $product->checkQty(1) return InStock;" before this?

File: classes/models/NostoProduct.php [hunk 2]
```diff
@@ -255,7 +255,16 @@ protected static function checkAvailability(Product $product)
...
+            return $product->checkQty(1) ? self::IN_STOCK : self::OUT_OF_STOCK;
+        }
+        foreach ($combinations as $combination) {
+            if ($combination['quantity'] > 0) {
```
Comment (chenliangde) [inline classes/models/NostoProduct.php]: Is it possible that a combination is "inactive" or something?

File: classes/models/NostoProduct.php (hunk 1 above)
Comment (supercid) [reply classes/models/NostoProduct.php]: You mean like this?
```php
        if ($product->checkQty(1)) {
            return self::IN_STOCK;
        }
        $combinations = $product->getAttributeCombinations();
        if (empty($combinations)) {
            return $product->checkQty(1) ? self::IN_STOCK : self::OUT_OF_STOCK;
        }
```

File: classes/models/NostoProduct.php (hunk 2 above)
Comment (supercid) [reply classes/models/NostoProduct.php]: If the product has no SKU's, the combination array comes empty

**Pull request merged**